crypto/encode_decode/decoder_lib.c

```diff
@@ -646,7 +647,9 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
                 && ERR_GET_REASON(err) == EC_R_UNKNOWN_GROUP)
 #endif
             || (ERR_GET_LIB(err) == ERR_LIB_X509
-                && ERR_GET_REASON(err) == X509_R_UNSUPPORTED_ALGORITHM))
+                && ERR_GET_REASON(err) == X509_R_UNSUPPORTED_ALGORITHM)
+            || (ERR_GET_LIB(err) == ERR_LIB_PROV
+                && ERR_GET_REASON(err) == PROV_R_BAD_DECRYPT))
```

Comment:
This is problematic.  What of other providers, using codes that are a) allocated dynamically and whose reason codes you do not know at all?

Reply:
I rather think that the ERR filter idea might be the better thing to do.  This is an entanglement that is becoming difficult to fend off, isn't it?

Reply:
I see your point (which is similar to what we discussed earlier on a related topic).
I don't see how to apply an ERR filter, but I think in this case there should be a simple way out:
I've replaced the dependency on `PROV_R_BAD_DECRYPT` by `PKCS12_R_PKCS12_CIPHERFINAL_ERROR`,
which is also from the crypto lib, so independent of changes/additions to providers.

Reply:
> I don't see how to apply an ERR filter

We don't have that mechanism, it needs being invented first...

Reply:
> I've replaced the dependency on `PROV_R_BAD_DECRYPT` by `PKCS12_R_PKCS12_CIPHERFINAL_ERROR`,
which is also from the crypto lib, so independent of changes/additions to providers.

Yeah, the saving grace here is that *our* provider OSSL_STORE implementation doesn't handle PKCS#12 files on its own, just dumps the DER back to libcrypto.  This is still a hack, but it's at least manageable.